src/Framework/Blocks/BlockCollection.php

```diff
@@ -79,24 +79,107 @@ public function getBlocks(): array
     /**
      * @unreleased
      *
-     * @param BlockModel[] $blocks
-     * @param string $name
-     * @return BlockModel|null
+     * @return BlockCollection|null
      */
-    public function findByName(string $name, array $blocks = null)
+    public function findBlock(string $blockName, int $blockIndex = 0, BlockCollection $blockCollection = null, string $return = 'self', int &$count = 0)
```

Comment:
Are all these extra params useful or meaningful as a public api of block collection? To me, everything except name & index seems very specific for the _other_ methods in the collection to work.  

If i'm using `findBlock` (or preferably `findByName`) its most likely just to get the block attributes.

It makes me think we should have a protected property like `findBlockRecursive()` (or `findByNameRecursive()` since we're searching by name)  that's used internally with all the extra logic and a public method `findByName()` that's more simple and could even use the recursive one under the hood if need be.

Reply:
I've been thinking about that too. Will quickly change it.

Reply:
Resolved b64be7ae8eaaaf93cca8f0898162979ccfc1705f